core/src/main/scala/org/http4s/headers/Host.scala

```diff
@@ -17,14 +17,30 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import org.http4s.internal.parsing.{Rfc3986, Rfc7230}
 import org.http4s.util.Writer
+import scala.util.Try
 
 object Host extends HeaderKey.Internal[Host] with HeaderKey.Singleton {
   def apply(host: String, port: Int): Host = apply(host, Some(port))
 
   override def parse(s: String): ParseResult[Host] =
-    HttpHeaderParser.HOST(s)
+    ParseResult.fromParser(parser, "Invalid Host")(s)
+
+  //  // Do not accept scoped IPv6 addresses as they should not appear in the Host header,
+  //  // see also https://issues.apache.org/bugzilla/show_bug.cgi?id=35122 (WONTFIX in Apache 2 issue) and
+  //  // https://bugzilla.mozilla.org/show_bug.cgi?id=464162 (FIXED in mozilla)
+  private[http4s] val parser = {
+    val port = Parser.string(":") *> Rfc3986.digit.rep.string.mapFilter { s =>
+      Try(s.toInt).toOption
+    }
+
+    val host = Rfc7230.token.orElse(Uri.Host.ipLiteral.string).backtrack <* Rfc7230.ows
```

Comment:
not sure if this is correct.

Ref comment above, Ipv6 should be disallowed, and this does not include ipv4 ?

Reply:
``` private[http4s] val parser = {
    val host =
      (Uri.Ipv4Address.parser).orElse(Uri.RegName.parser)
    val port = Parser.string(":") *> Rfc3986.digit.rep.string.mapFilter { s =>
      Try(s.toInt).toOption
    }

    (host ~ port.?).map { case (host, port) =>
      Host(host.value, port)
    }
  }```
Possible implementation

Reply:
Hmm you are absolutely right. Weird thing is that I'm pretty sure this is than wrong in 0.21

Reply:
https://github.com/http4s/http4s/pull/4208/commits/69051b341fba89b474cb27fa12f83755758f458f